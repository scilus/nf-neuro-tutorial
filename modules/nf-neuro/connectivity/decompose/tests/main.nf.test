nextflow_process {

    name "Test Process CONNECTIVITY_DECOMPOSE"
    script "../main.nf"
    process "CONNECTIVITY_DECOMPOSE"

    tag "modules"
    tag "modules_nfcore"
    tag "connectivity"
    tag "connectivity/decompose"

    tag "subworkflows/load_test_data"

    setup {
        run("LOAD_TEST_DATA", alias: "LOAD_DATA") {
            script "../../../../../subworkflows/nf-neuro/load_test_data/main.nf"
            process {
                """
                input[0] = Channel.from( [ "connectivity.zip" ] )
                input[1] = "test.load-test-data"
                """
            }
        }
    }

    test("connectivity - decompose - without option") {
        when {
            process {
                """
                input[0] = LOAD_DATA.out.test_data_directory.map{
                    test_data_directory -> [
                        [ id:'test', single_end:false ], // meta map
                        file("\${test_data_directory}/bundle_all_1mm.trk", checkIfExists: true),
                        file("\${test_data_directory}/endpoints_atlas.nii.gz", checkIfExists: true)
                    ]
                }
                """
            }
        }
        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }

        test("connectivity - decompose - with all option") {
        config "./nextflow.config"
        when {
            process {
                """
                input[0] = LOAD_DATA.out.test_data_directory.map{
                    test_data_directory -> [
                        [ id:'test', single_end:false ], // meta map
                        file("\${test_data_directory}/bundle_all_1mm.trk", checkIfExists: true),
                        file("\${test_data_directory}/endpoints_atlas.nii.gz", checkIfExists: true)
                    ]
                }
                """
            }
        }
        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }

        test("connectivity - visualisation - stub-run") {
        options "-stub-run"
        when {
            process {
                """
                input[0] = LOAD_DATA.out.test_data_directory.map{
                    test_data_directory -> [
                        [ id:'test', single_end:false ], // meta map
                        file("\${test_data_directory}/bundle_all_1mm.trk", checkIfExists: true),
                        file("\${test_data_directory}/endpoints_atlas.nii.gz", checkIfExists: true)
                    ]
                }
                """
            }
        }
        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out.versions).match() }
            )
        }
    }
}
