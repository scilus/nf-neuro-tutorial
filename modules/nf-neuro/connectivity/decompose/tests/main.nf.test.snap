{
    "connectivity - decompose - without option": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test__decomposed.h5:md5,90ff9ae7ceb1371349231f85b616874e"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test__labels_list.txt:md5,3b0332e02daabf31651a5a0d81ba830a"
                    ]
                ],
                "2": [
                    "versions.yml:md5,ca6d20bf86c1e32c1bccea5813810f02"
                ],
                "hdf5": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test__decomposed.h5:md5,90ff9ae7ceb1371349231f85b616874e"
                    ]
                ],
                "labels_list": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test__labels_list.txt:md5,3b0332e02daabf31651a5a0d81ba830a"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,ca6d20bf86c1e32c1bccea5813810f02"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.10.3"
        },
        "timestamp": "2025-01-16T12:03:16.31875384"
    },
    "connectivity - visualisation - stub-run": {
        "content": [
            [
                "versions.yml:md5,ca6d20bf86c1e32c1bccea5813810f02"
            ]
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.10.3"
        },
        "timestamp": "2025-01-16T12:03:27.09566413"
    },
    "connectivity - decompose - with all option": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test__decomposed.h5:md5,5660bb84f333b6586ed247c6b95f6a98"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test__labels_list.txt:md5,3b0332e02daabf31651a5a0d81ba830a"
                    ]
                ],
                "2": [
                    "versions.yml:md5,ca6d20bf86c1e32c1bccea5813810f02"
                ],
                "hdf5": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test__decomposed.h5:md5,5660bb84f333b6586ed247c6b95f6a98"
                    ]
                ],
                "labels_list": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test__labels_list.txt:md5,3b0332e02daabf31651a5a0d81ba830a"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,ca6d20bf86c1e32c1bccea5813810f02"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.10.3"
        },
        "timestamp": "2025-01-16T12:03:21.738887954"
    }
}